### DISCUSSION TITLE
HSMM representations

### DISCUSSION THREAD

**Discussion by riccile** [bmcclintock/momentuHMM#53] [Q&A] [answered]
Hi Brett,
I'm trying to run a HSMM by expanding the number of states in my HMM. I'm hoping to go from a 4 state model to an 8 state model (3 states with geometric distributions and 5 states that expand the last state with dwell times pulled from a negative binomial distribution. 

I made the expanded transition matrix following Zucchini's advice but I'm not sure how to incorporate it into the model. I believe I can fix transitions probabilities to 0 with fixPar$beta0 but I'm not sure how to fix the transition probabilities based on the dwell-time distribution. 

Any help would be appreciated. Thanks!

**Comment by bmcclintock** [comment, marked as answer]
Good question! I've thought a little bit about implementing HSMMs with `momentuHMM` in the past, and, while I believe non-parametric dwell-time distributions could be implemented using `fixPar$beta`, I'm not so sure about parametric dwell-time distributions. It would likely be tedious, but I believe non-parametric dwell-time distributions could be accommodated by adding ![formula](https://render.githubusercontent.com/render/math?math=m_j) "micro-states" (each with no self transition, where ![formula](https://render.githubusercontent.com/render/math?math=m_j) is an upper bound for the maximum time spent in state j before a transition between "macro-states" can potentially occur) for each macro-state j=1,...,N (N=8 in your case). Those macro-states with geometric dwell times would of course have ![formula](https://render.githubusercontent.com/render/math?math=m_j=1). This structure might be more easy to set up using the hierarchical HMM functions (see `?momentuHMM:::fitHMM.momentuHierHMMData`, `?momentuHMM::simHierData`, and examples in the [vignette](https://CRAN.R-project.org/package=momentuHMM/vignettes/momentuHMM.pdf)). However, the resulting TPM would be of dimension ![formula](https://render.githubusercontent.com/render/math?math=\sum_{j=1}^Nm_j\times\sum_{j=1}^Nm_j) and rather sparse, which could pose computational issues and be very slow.

The problem with trying to shoehorn parametric dwell-time distributions into `momentuHMM` is that the working-scale `covs %*% beta` matrix is transformed to the real-scale `gamma` using a multinomial logit link. One could potentially use `fixPar$beta` to set the micro-state transitions according to a (known) dwell-time distribution, but I don't see how one could estimate the parameters of a parametric dwell-time distribution using `momentuHMM`. If you really want to fit (parametric) HSMMs, it may be worth exploring the [`hsmm`](https://CRAN.R-project.org/package=hsmm) and [`mhsmm`](https://CRAN.R-project.org/package=mhsmm) packages.
